src/sidecar/connectionStatusUtils.ts

```diff
@@ -0,0 +1,50 @@
+// helpers for connection status testing, factored out for test spying
```

Comment:
These functions used to be inlined within the while loop of the original spelling of `waitForConnectionToBeUsable`. Now extracted with isConnectionStable used as a predicate function for `waitForConnectionToBeStable`'s interaction with `ConnectionStateWatcher.waitForConnectionUpdate()` and now unit-testable